classes/class.memberorder.php

```diff
@@ -142,8 +158,11 @@ function getMemberOrderByID($id)
 				$this->checkout_id = $dbobj->checkout_id;
 
 				//reset the gateway
-				if(empty($this->nogateway))
+				if( empty( $this->nogateway ) ) {
 					$this->setGateway();
+				} else {
+					unset( $this->Gateway );
```

Comment:
Must `unset` to be sure the `__get` is called. This should be safe to be done every time the ->nogateway is specified, but probably useless.